### PR TITLE
Db migration app from mainnet to berkeley

### LINKED ISSUE #10945: Migration tool for mainnet DB post-fork

### LINKED ISSUE #12676: Plan for devnet and mainnet archive DB migration

### PR DESCRIPTION
**Pull request by psteckler**
App to migrate mainnet archive db data to berkeley schema. Ledger hashes and snarked ledger hashes are left intact. The implementation converts each block into the "extensional" format, and imports that into the new db.

Notes:
- the transaction hashes are passed through unaltered
- the app will resume where it left off, allowing incremental migration
- the `accounts_accessed` table is not populated, that requires the ledger contents; a modified replayer will do that in a separate migration step

Part of #12676.

Part of #14313.

Closes #10945.

### CODE REVIEW

**Comment by psteckler** [comment]
!ci-build-me

**Comment by psteckler** [comment]
!ci-build-me

**Review by MartinMinkov — APPROVED** [review]

**Comment by psteckler** [comment]
!ci-build-me

File: src/app/archive/lib/processor.ml [hunk 1]
```diff
@@ -3233,19 +3235,15 @@ module Block = struct
...
-                      ledger_hash, height,
-                      global_slot_since_hard_fork, global_slot_since_genesis,
-                      protocol_version, proposed_protocol_version,
-                      timestamp, chain_status)
-                      VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?::bigint[], ?, ?, ?, ?, ?, ?, ?, ?, ?::chain_status_type)
-                     RETURNING id
-               |sql} )
+               (Mina_caqti.insert_into_cols ~returning:"id" ~table_name
+                  ~tannot:(function
+                    | "sub_window_densities" ->
```
Comment (ghost-not-in-the-shell) [inline src/app/archive/lib/processor.ml:3240]: Would this change affect the archiver on berkeley branch?

Comment (psteckler) [reply src/app/archive/lib/processor.ml:3240]: Do you mean the `berkeley` testnet (this PR is against the `berkeley` branch)?

This fix affects only archiving extensional blocks, so it doesn't affect the processor running on the testnet.

**Review by ghost-not-in-the-shell — APPROVED** [review]

**Comment by psteckler** [comment]
!ci-build-me

**Comment by deepthiskumar** [comment]
!approved-for-mainnet

**Pull request merged**